### ISSUE TITLE
Cloud testing: Spike on credentials and workflows

### ISSUE THREAD

**Issue by algattik** [agera-edc/DataSpaceConnector#146]
Feature #285 

## Description

In #115 we need to run integration tests on actual cloud resources (Data Factory) for which no emulators exist.
To simplify that story, the tests are run as follows:
- In upstream repo, the integration test only runs after merge.
- People who submit PRs should run the pipeline in their own fork and subscription

This is good enough for a start, but not ideal, because reviewers cannot be fully sure that the submitted code really will not break the pipeline. In the future we want something more robust.

An upstream committer should be able to run the Azure test pipeline on demand, after having inspected that the code has no security risk (secret leakage). Various ways for doing this exist:
- writing a bot application, but that adds complexity
- triggering on a PR label, but that is not safe as the author could add malicious code afterward to re-trigger the pipeline (to be checked)
- triggering on a PR comment

Proposal for a spike:
- Use a workflow triggered by a PR comment (e.g. `/test`)
- Run tests with a workflow triggered by `workflow_call`, that is run both by the PR comment workflow and the standard CI workflow
- Use of [Federated access](https://docs.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Cwindows)
- [Dismiss stale approvals](https://stackoverflow.com/questions/40505904/is-there-a-way-to-make-github-un-approve-a-pull-request-if-a-new-commit-is-pushe)

Out of scope:
- Writing a GitHub bot (an externally hosted application)
- Automating Terraform deployment

## Acceptance Criteria

- [ ] ADR 
- [ ] Demo for the EDC community (Jim/Paul)
- [ ] An upstream committer can to run the Azure test pipeline on demand, after having inspected that the code has no security risk
- [ ] Scenario where the PR author modifies the PR after tests are run is explored, also considering that code might be malicious
- [ ] Follow-up implementation story #192 refined and ready
- [ ] Follow-up sharing artifact creation story created

## Tasks

- [ ] Contact Markus for the Cosmos related secrets creation (when he has access)
- [ ]

**Comment by chrislomonico** [comment]
ADF configuration needs to be in place prior to starting this work.  In the meantime; reach out to Markus to get the secrets set in GH

**Comment by zeier** [comment]
To be postponed for later, depending on whether there is a big enough need for this.
